redi-connect-front/src/pages/app/me/PersonalDetail.tsx

```diff
@@ -63,11 +63,9 @@ const PersonalDetail = ({ profile, profileSaveStart }: any) => {
   })
 
   const readPersonalDetail = () => {
-    // TODO: check if there is a nicer soulution with bulma as the profile__list
-    return <Content className="profile__list">
-      {gender && <span>{formGenders.filter(formGender => formGender.value === gender).map(formGender => formGender.label) }</span>}
-      {age && <span>{`${age} years old`}</span>}
-    </Content>
+    const detailsList = [gendersIdToLabelMap[gender]]
```

Comment:
Let's avoid defining inline components inside of a component like this. Remember the trouble we had earlier this week with the animation, mounting/unmounting?

I think this can be made more elegant by:

First, moving this code to right below the closing curly brace of `readPersonalDetail`:
```
const detailsList = [gendersIdToLabelMap[gender]]
if (age) detailsList.push(`${age} years old`)
```

Second, moving this directly into the `read` parameter:
```
read={emptyProfile && <PipeList items={detailsList} />}
```
(this pattern of initiating a component inside of a prop is commonplace.

And thirdly, getting rid of the whole `readPersonalDetail` function

Reply:
I've seen this pattern used for the `read` prop  elsewhere, too. Let's refactor those ones as well.

Reply:
Good idea. thanks